app/controllers/api/v1/products_controller.rb

```diff
@@ -1,7 +1,8 @@
 class Api::V1::ProductsController < Api::V1::BaseController
+  before_action :intialize_category
 
   def index
-    products = Product.all
+    products = @category.products.all
     render_with_options(
```

Comment:
What's the benefit of using `render_with_options`? (just curios) as `render json: @category.products` should do the work :)

Reply:
As seen here we can specify which serializer to use, suppose if we multiple serializers for different actions of products, with the help of this method we can name them.